cpp/src/arrow/compute/exec/test_util.cc

```diff
@@ -459,5 +460,44 @@ void PrintTo(const Declaration& decl, std::ostream* os) {
   *os << "}";
 }
 
+std::shared_ptr<Table> MakeRandomTable(TableGenerationProperties properties) {
+  int total_columns = properties.num_columns + 2;
+  std::vector<std::shared_ptr<Array>> columns;
+  columns.reserve(total_columns);
+  arrow::FieldVector field_vector = arrow::FieldVector();
+  field_vector.reserve(total_columns);
+
+  field_vector.push_back(std::make_shared<Field>("time", int64()));
+  field_vector.push_back(std::make_shared<Field>("id", int32()));
```

Comment:
```suggestion
  field_vector.push_back(field("time", int64()));
  field_vector.push_back(field("id", int32()));
```

What you have is fine, this is just a slightly more compact shortcut